sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/backoff/BackOff.java

```diff
@@ -0,0 +1,74 @@
+package com.mesosphere.sdk.scheduler.plan.backoff;
+
+import com.mesosphere.sdk.framework.EnvStore;
+import com.mesosphere.sdk.offer.LoggingUtils;
+
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Abstract class that has contracts on how delay can be set/advanced/cleared. Helper methods are used to
+ * instantiate either a {@link ExponentialBackOff} or {@link DisabledBackOff} based on environment.
+ */
+public abstract class BackOff {
+  private static final String FRAMEWORK_BACKOFF_FACTOR = "FRAMEWORK_BACKOFF_FACTOR";
+
+  private static final String FRAMEWORK_INITIAL_BACKOFF = "FRAMEWORK_INITIAL_BACKOFF";
+
+  private static final String FRAMEWORK_MAX_LAUNCH_DELAY = "FRAMEWORK_MAX_LAUNCH_DELAY";
+
+  private static final String ENABLE_BACKOFF = "ENABLE_BACKOFF";
+
+  private static final Object lock = new Object();
+
+  private static Logger logger = LoggingUtils.getLogger(BackOff.class);
+
+  private static volatile BackOff instance;
+
+  /**
+   * Backoff is enabled only by explicit Opt-In. We adhere to following configuration:
+   *  * Set {@link #ENABLE_BACKOFF} to non blank value.
+   *  * Set one or more of {@link #FRAMEWORK_BACKOFF_FACTOR} or
+   *    {@link #FRAMEWORK_INITIAL_BACKOFF} and/or {@link #FRAMEWORK_MAX_LAUNCH_DELAY}.
+   *    If more than zero params are initialized, defaults are used for the uninitialized params.
+   *  * Absence of {@link #ENABLE_BACKOFF} and other environment params would be considered
+   *    as hint to disable backoff
+   */
+  public static BackOff getInstance() {
+    if (instance == null) {
+      synchronized (lock) {
+        EnvStore envStore = EnvStore.fromEnv();
+        if (envStore.isPresent(ENABLE_BACKOFF)
+                || envStore.isPresent(FRAMEWORK_BACKOFF_FACTOR)
```

Comment:
I don't think we need to check for these three env vars, the latter call uses `envStore.getOptional*` which means its enough to rely on the enable/disable flag.